### DISCUSSION TITLE
Can't open react app in code-server while running on docker container.

### DISCUSSION THREAD

**Discussion by Avirup-301294** [coder/code-server#5488] [Q&A] [answered]
I tried running code-server in windows, so I used docker command.

`mkdir -p ~/.config
docker run -it --name code-server -p 127.0.0.1:8080:8080 \
  -v "$HOME/.config:/home/coder/.config" \
  -v "$PWD:/home/coder/project" \
  -u "$(id -u):$(id -g)" \
  -e "DOCKER_USER=$USER" \
  codercom/code-server:latest`

After that I tried creating an react app using **npx create-react-app myapp**. After creating the app, I issue npm start to start the app.Till now everything was fine, but I am not able to see the application on localhost:3000. It is obvious that the localhost is not seen outside container. Also I searched for docker networking and found if I put --network="host" it will in case of Linux, but not on windows and Mac.

So, here is my question, how will I use same host network feature in windows without any proxy like 3000-mydomain-proxy. I want to run my react on particular ip address. Kindly suggest solution.

**Comment by jsjoeio** [comment, marked as answer]
Hmm....so outside a docker container, localhost works so localhost:3000 works, which we know.

Do you need an additional `-p 127.0.0.1:3000:3000` in the `run` command? But then if you change the port, you'd have to kill the container not ideal. 

I found [this](https://stackoverflow.com/a/33699823/3015595):

```shell
docker run -p 2000-5000:2000-5000 -v /host/:/host appimage
```

to expose a range of ports.

cc @code-asher in case you know other solutions

**Reply by code-asher** [reply]
The only alternative I can think of is to access the React app through code-server's path-based proxy: `http://localhost:8080/absproxy/3000/` but some additional configuration options may be needed for create-react-app: https://github.com/coder/code-server/blob/main/docs/guide.md#proxying-to-create-a-react-app

**Reply by code-asher** [reply]
Ah the original post says without any proxy.  In that case the additional `-p 3000:3000` is the best solution (or a range if you anticipate using many ports).

**Reply by Avirup-301294** [reply]
Thanks, @jsjoeio @code-asher it works nicely! :)
